### ISSUE TITLE
bug: Incompatible with griffe v1.0

### ISSUE THREAD

**Issue by glennmatthews** [mkdocstrings/python#181]
### Description of the bug

Readthedocs build started failing with mkdocstrings-python v1.10.7 and mkdocstrings v0.25.2:

```
  File "/home/docs/checkouts/readthedocs.org/user_builds/nautobot/envs/6107/lib/python3.9/site-packages/mkdocstrings_handlers/python/handler.py", line 15, in <module>
    from griffe import (
ImportError: cannot import name 'patch_logger' from 'griffe' (/home/docs/checkouts/readthedocs.org/user_builds/nautobot/envs/6107/lib/python3.9/site-packages/griffe/__init__.py)
```

This appears to be because griffe v1.0 was [released](https://github.com/mkdocstrings/griffe/releases/tag/1.0.0) and contains breaking API changes, but mkdocstrings-python has no upper bound on its permitted `griffe` dependency that would prevent this incompatible upgrade:

```
    "griffe>=0.49",
```

**Comment by pawamoy** [comment]
Ah, obviously I had to forget the most important update. Lemme fix fix asap.

**Comment by pawamoy** [comment]
Actually could you try to upgrade mkdocstrings-python? Latest version1.10.8 should use `patch_loggers`, as expected by Griffe v1.

**Comment by glennmatthews** [comment]
I'll test updating to 1.10.8 on Monday unless someone else on our end beats me to it, but might I suggest that prior releases of this package probably should have specified `griffe>=...,<1.0` so that the new release wouldn't break previously working deployment setups? 20/20 hindsight. :-)

**Comment by glennmatthews** [comment]
Had a few minutes this afternoon to test, and it does look like 1.10.8 works fine with griffe 1.0. Thanks!

**Comment by pawamoy** [comment]
Technically, I think only 1.10.7 is incompatible. The plan was always to have backward-compatible releases until Griffe v1. Python citizens usually know that it's best to avoid upper bounds, as unfortunate breakages can always be avoided downstream, while upstream constraints cannot be cancelled.

Also, dependency upgrade processes should probably upgrade everything at once (maintaining compatible versions), not just one package at a time.

Long story short: I do my best to keep things working, all you have to do is stay up to date with each dependency ^^

**Comment by pawamoy** [comment]
But yeah, the `patch_logger`/`patch_loggers` is my mistake (see Griffe's changelog) ^^'
